src/modules/CC1101/CC1101.cpp

```diff
@@ -268,31 +294,66 @@ int16_t CC1101::readData(uint8_t* data, size_t len) {
     SPIreadRegister(CC1101_REG_FIFO);
   }
 
-  // read packet data
-  SPIreadRegisterBurst(CC1101_REG_FIFO, length, data);
+  uint8_t bytesInFIFO = SPIgetRegValue(CC1101_REG_RXBYTES, 6, 0);
+  uint16_t readBytes = 0;
+  unsigned long lastPop = millis();
+
+  // keep reading from FIFO until we get all the packet.
+  while (readBytes < length) {
+    if (bytesInFIFO == 0) {
+      if (millis() - lastPop > 5) {
+        // readData was required to read a packet longer than the one received.
+        RADIOLIB_DEBUG_PRINT(F("No data for more than 5mS. Stop here."));
+        break;
+      } else {
+        delay(1);
```

Comment:
Similar to one of the previous comments - will this delay work on high data rates?